src/viam/sdk/components/private/board_server.cpp

```diff
@@ -174,6 +177,51 @@ ::grpc::Status BoardServer::GetDigitalInterruptValue(
     return ::grpc::Status();
 }
 
+::grpc::Status BoardServer::StreamTicks(
+    ::grpc::ServerContext* context,
+    const ::viam::component::board::v1::StreamTicksRequest* request,
+    ::grpc::ServerWriter<::viam::component::board::v1::StreamTicksResponse>* writer) {
+    if (!request) {
+        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
+                              "Called [Board::StreamTicks] without a request");
+    };
+
+    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
+    if (!rb) {
+        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
+    }
+
+    const std::shared_ptr<Board> board = std::dynamic_pointer_cast<Board>(rb);
+
+    AttributeMap extra;
+    if (request->has_extra()) {
+        extra = struct_to_map(request->extra());
+    }
+
+    const std::shared_ptr<std::queue<Board::tick>> ticks;
+    const std::vector<std::string> digital_interrupt_names(request->pin_names().begin(),
+                                                           request->pin_names().end());
+
+    board->stream_ticks(digital_interrupt_names, ticks, extra);
+
+    ::viam::component::board::v1::StreamTicksResponse response;
+    while (true) {
```

Comment:
Is there any risk that this will become blocking on any other calls to a board server? Should we be putting this loop in a separate thread and returning right away?

Reply:
`stream_ticks` should continue running continuously until cancelled. I'm not sure if this will block other calls to the server, can you not have multiple server calls running at the same time?

Reply:
Unfortunately, I don't think the API as currently designed is going to work with that model. I will leave additional comments on `board.hpp` about why.